### PR TITLE
New Config file to change settings easily, New security checks

### PR DESCRIPTION
**Pull request by devtekve**
Addes a new Config File to easily set up billingserver and set database
preferences and add allowed ip/hosts

The system will perfom a check of the IP or Host before making any
response to see if client is allowed.

### CODE REVIEW

**Comment by florian0** [comment]
Nice! Thank you.
There is one little bug. The prefix of the `HttpListener` (https://github.com/devtekve/MiniBillingServer/blob/c648a01072f3e3c068ef7a7abb686fc9cdca9e4a/Program.cs#L21) defines the hostname on which the server is accepting requests (this is the Host-field in the HttpRequest-Header). When allowing restricted external access to the server, the hostname can neither be `127.0.0.1` nor `localhost`, since an external host will not access the server using one of these.

Long story short: The hostname/prefix should be configurable aswell. Would be great if you could add that. You also might change the related readme line explaining what to put as billing server url.

**Comment by devtekve** [comment]
Everything should be fine now
